### DISCUSSION TITLE
How to create indexers on my own Python code?

### DISCUSSION THREAD

**Discussion by NathanAP** [microsoft/graphrag#983] [Q&A]
Hi all,

I'm trying to use GraphRag on my own Python code to create indexed text documents. 

I was able to reproduce it through the Indexer CLI and worked perfectly, but I want to execute it using my own logic in my code instead of using `python -m ...` on command line.

Is there an example I can follow to reach it?

edit: not really sure I was clear enough. I want to avoid, for example, executing it using `subprocess`.

**Comment by giu-ferreira-cientista** [comment]
I need this too! Any code example?

**Comment by l4b4r4b4b4** [comment]
I would do the following:
1. Put your code as txt in `input`
2. Define custom entities
3. Define custom promptOptimisation

**Comment by MisterAndry** [comment]
https://github.com/microsoft/graphrag/discussions/694
